spec/core/link.fmf

```diff
@@ -0,0 +1,107 @@
+summary: Link related objects
+
+story:
+    As a user I want to link related tests, stories or external
+    issues so that I can easily find out what the test, plan or
+    story is addressing.
+
+description: |
+    The core attribute ``link`` is used to track relevant objects
+    such as user stories or external issues verified by the given
+    test, source code implementing particular story, blocking
+    issues or parent-child relations.
+
+    The value can be a single link or a list of links. There are
+    two options how to specify the ``target`` of the reference:
+
+    string
+        This can be either a ``name`` of the fmf node within the
+        same metadata tree, a file ``path`` from the tree root to
+        the application or documentation source code or a ``URL``
+        to an external issue tracker.
+
+    dict
+        In order to reference remote fmf objects use a dictionary
+        with the full `fmf identifier`__.
+
+    By default, the link between the two objects is interpreted in
+    a generic way as ``relates``. To explicitly define a different
+    connection to the target reference, use a dictionary and set
+    the key to one of the supported relations:
+
+    verifies
+        This object (a test or a plan) verifies functionality
+        defined in the target (a story, an issue, a bug).
+
+    verified-by
+        The functionality (described in a story, an issue or a
+        bug) is verified by the target (a test or a plan).
+
+    implements
+        This object (source code) implements functionality defined
+        in the target (a story describing the feature).
+
+    implemented-by
+        The functionality (described in a story) is implemented by
+        the target (source code).
+
+    documents
+        This object (a guide or examples) documents the target (a
+        story).
+
+    documented-by
+        The functionality (described in a story) is documented by
+        the target (a guide or examples).
+
+    blocks
+        This object is blocking the target.
+
+    blocked-by
+        This object is blocked by the target.
+
+    duplicates
+        This object is a duplicate of the target.
+
+    duplicated-by
+        This object is duplicated by the target.
+
+    parent
```

Comment:
(also `child`) it is not described well. not easy to understad what `target` means (intuitively, for me it means something like merging of FMF (sub-)trees.

Reply:
The `target` term is defined at the top of the spec and is consistently used across all other relation definitions. Any suggestion how to make the term even more clear?

Reply:
yes, it is there defined that it means some operation with FMF tree (local or remote) but from my perspective, it is not now supported to merge metadata trees in FMF so that I do not know how it should behave.
if this `parent`/`child` means (move or merge ) the current node to another location in sense of operations like
```
this_node.parent = tree.find(STR)
tree.find(STR).children[this_node.name] = this_mode
tree.find(STR).update()
```

Reply:
> it is there defined that it means some operation with FMF tree

No, as for now there is no operation (such are merging or grafting metadata trees) planned to be done. The spec only defines how to store target reference (as a string or as a dictionary). In the future we can extend the API to do some object linking magic as well but this is not in the scope of this pull request.